[Python] Inventory Dicts Improvement (#2770)
exercism/v3: languages/python/exercises/concept/inventory-management/.docs/introduction.md

Dictionaries can be created in various ways. Two simple options are the use the `dict()` class constructor or the dict literal declaration with key-value pairs.

### Use the `dict()` constructor

```python
>>> bear = dict(name="Black Bear", speed=40, land_animal=True)
{'name': 'Black Bear', 'speed': 40, 'land_animal': True}
```

### Declare a _dict_ literal

```python
>>> whale = {"name": "Blue Whale", "speed": 35, "land_animal": False}
{'name': 'Blue Whale', 'speed': 35, 'land_animal': False}
```

With the dict literal declaration keep in mind that _keys_ are of _data types_ `str` and the colon `:` is used instead of an equal sign `=`.

## Accessing values

You can access an item in a dictionary using the _key_ of the value.

### Using _square brackets_ after the dict object

```python
>>> bear["speed"]
40
```

### Using `.get()`

```python
>>> whale.get("name")
'Blue Whale'
```

## Changing values

You can easily change a value of an item using its _key_.

```python
>>> bear["name"] = "Grizzly Bear"
{'name': 'Grizzly Bear', 'speed': 40, 'land_animal': True}

>>> whale["speed"] = 25
{'name': 'Blue Whale', 'speed': 25, 'land_animal': False}
```

## Looping through a dictionary

> Looping through a dictionary using a `for` loop only returns the _keys_.

Reviewer: ```suggestion
Looping through a dictionary using `for item in` will default to iterating over the _keys_.
```

```python
>>> for key in bear:
>>>     key
'name'
'speed'
'land_animal'
```

> But you can also access the _values_ of a dictionary using _square brackets_.

Reviewer: ```suggestion
But you can also access the _values_ of a dictionary by using _square brackets_ after each key:
```
